### PR TITLE
Add in max_sub_tokens support

### PR DESCRIPTION
**Pull request by derekcollison**
Signed-off-by: Derek Collison <derek@nats.io>

/cc @nats-io/core

### CODE REVIEW

File: server/client.go [hunk 1]
```diff
@@ -2418,6 +2418,14 @@ func (c *client) processSubEx(subject, queue, bsid []byte, cb msgHandler, noForw
 			c.subPermissionViolation(sub)
 			return nil, ErrSubscribePermissionViolation
 		}
+
+		if maxSubTokens := int(c.srv.getOpts().MaxSubTokens); maxSubTokens > 0 {
```
Comment (kozlovic) [inline server/client.go:2422]: `srv` has already been captured as `c.srv`, but no biggy.

**Review by kozlovic — CHANGES_REQUESTED** [review]
I saw fix for reload, but that's not really complete: if we support, then we need to add new object and report that it has been changed during reload, and decide what we do with existing subs? Or fail the reload if the value is changed?

File: server/client.go (hunk 1 above)
Comment (derekcollison) [reply server/client.go]: fixed

**Comment by derekcollison** [comment]
I think on reload it just enforces going forward, no changes imo. The ones that failed before of course aren't there and the ones that succeeded but may have failed with the new restriction I think we leave them.

**Review by kozlovic — APPROVED** [review]
LGTM - But it means that the server is not going to print that it has reloaded (and accepted) the new value, but that is not crucial I guess.

**Comment by kozlovic** [comment]
That's fine, we can add later.

**Pull request merged**